client/my-sites/email/email-pricing-notice/index.tsx

```diff
@@ -0,0 +1,176 @@
+	}
+
+	if ( ! purchaseCost ) {
+		return true;
+	}
+
+	return (
+		purchaseCost.amount === mailProduct.cost && purchaseCost.currency === mailProduct.currency_code
+	);
+};
+
+function getPriceMessage( {
+	mailboxPurchaseCost,
+	translate,
+}: {
+	mailboxPurchaseCost: EmailCost | null;
+	translate: typeof originalTranslate;
+} ): TranslateResult {
+	if ( mailboxPurchaseCost === null ) {
+		return '';
+	}
+	return mailboxPurchaseCost.amount === 0
+		? translate( 'You can add new mailboxes for free until the end of your trial period.' )
+		: translate(
+				'You can purchase new mailboxes at the prorated price of {{strong}}%(proratedPrice)s{{/strong}} per mailbox.',
+				{
+					args: {
+						proratedPrice: mailboxPurchaseCost.text,
+					},
+					components: {
+						strong: <strong />,
+					},
+					comment:
+						'%(proratedPrice)s is a formatted price for an email subscription (e.g. $3.50, €3.75, or PLN 4.50)',
+				}
+		  );
+}
+
+function getPriceMessageExplanation( {
+	mailboxPurchaseCost,
+	mailboxRenewalCost,
+	translate,
+}: {
+	mailboxPurchaseCost: EmailCost | null;
+	mailboxRenewalCost: EmailCost | null;
+	mailProduct: ProductListItem;
+	translate: typeof originalTranslate;
+} ): TranslateResult {
+	if ( mailboxPurchaseCost === null || mailboxRenewalCost === null ) {
+		return '';
+	}
+
+	// We don't need any explanation of the price at this point, because we have already handled it previously.
+	if ( mailboxPurchaseCost.amount === 0 ) {
+		return '';
+	}
+
+	if ( mailboxPurchaseCost.amount < mailboxRenewalCost.amount ) {
+		return translate(
+			'This is less than the regular price because you are only charged for the remainder of the current year.'
+		);
+	}
+
+	return translate(
+		'This is more than the regular price because you are charged for the remainder of the current year plus any additional year until renewal.'
+	);
+}
+
+function getPriceMessageRenewal( {
+	expiryDate,
+	mailboxRenewalCost,
+	translate,
+}: {
+	expiryDate: string;
+	mailboxRenewalCost: EmailCost | null;
+	translate: typeof originalTranslate;
+} ): TranslateResult {
+	if ( mailboxRenewalCost === null ) {
+		return '';
+	}
+
+	return translate(
+		'All of your mailboxes are due to renew at the regular price of {{strong}}%(fullPrice)s{{/strong}} per mailbox when your subscription renews on {{strong}}%(expiryDate)s{{/strong}}.',
+		{
+			args: {
+				fullPrice: mailboxRenewalCost.text,
+				expiryDate: expiryDate,
+			},
+			components: {
+				strong: <strong />,
+			},
+			comment:
+				'%(fullPrice)s is a formatted price for an email subscription (e.g. $3.50, €3.75, or PLN 4.50), ' +
+				'%(expiryDate)s is a localized date (e.g. February 17, 2021)',
+		}
+	);
+}
+
+interface MailboxPricingNoticeProps {
+	domain: ResponseDomain;
+	expiryDate: string | null;
+	mailProduct: ProductListItem;
+	mailboxPurchaseCost: EmailCost | null;
+	mailboxRenewalCost: EmailCost | null;
+}
+
+const MailboxPricingNotice = ( {
+	domain,
+	expiryDate,
+	mailProduct,
```

Comment:
I would suggest renaming `mailProduct` to `product` (or at least `emailProduct`).

Reply:
Makes a lot of sense :)